runtime/doc/lsp.txt

```diff
@@ -789,7 +789,10 @@ buf_request_sync({bufnr}, {method}, {params}, {timeout_ms})
     Parameters: ~
       • {bufnr}       (`integer`) Buffer handle, or 0 for current.
       • {method}      (`string`) LSP method name
-      • {params}      (`table?`) Parameters to send to the server
+      • {params}      (`table|(fun(client: vim.lsp.Client, bufnr: integer): table?)?`)
+                      Parameters to send to the server. Can also be passed as
+                      a function that returns the params table for cases where
+                      parameters are specific to the client.
```

Comment:
```suggestion
                      Parameters to send to the server. If a function, it is passed the LSP client and the buffer number and should return the respective parameters.
```

Reply:
Will reword to this now @MariaSolOs .

Just to point out, this docstring was a direct copy/paste of `buf_request_all`'s `params` arg [here](https://github.com/neovim/neovim/blob/5f9f5bc04d0bec0fbf100dd5a0de2b38daf882fb/runtime/lua/vim/lsp.lua#L1196-L1198), should I reword that too to be consistent?

Reply:
Ohh I see. I think that my suggested description is more helpful and we should indeed try to keep both consistent, but this is a biased opinion.

WDYT @lewis6991?

Reply:
Personally it doesn't seem necessary to redescribe the type annotation in English. It's clear what arguments the function accepts. This is unlike eval.txt which doesn't have type annotations, so descriptions of the arguments are necessary.

I don't mind either way though, but I'd lean on the side of being concise.

Reply:
Agreed. I kind of feel like the entire 2nd sentence isn't needed.

Reply:
OK so, remove everything after "Parameters to send to the server" then?

Reply:
I'm not sure if it's obvious enough that the function should return the request parameters as the return type is only `table`. I'm also not sure it's obvious that when it is passed as a function that the function itself is not the parameter, but the return of the function.

If you guys think it is then no worries.

Reply:
I personally think it is. But okay I made this PR more complicated than it should be lol.

@tigh-latte I think that using the same description as `buf_request_all` is fine (so what you originally had).

Reply:
Ha that's fine I'll move back to what was there originally

Reply:
done @MariaSolOs